tests/api_test.py

```diff
@@ -41,13 +41,29 @@
 from jax import test_util as jtu
 from jax import tree_util
 from jax import linear_util as lu
+from jax.lib import version
 
 from jax.config import config
 config.parse_flags_with_absl()
 FLAGS = config.FLAGS
 
 
-class JitTest(jtu.JaxTestCase):
+class CPPJitTest(jtu.JaxTestCase):
+  """Shared tests between the Python and the C++ jax,jit implementations.
```

Comment:
I might be tempted to structure this as a common test case class but to add skip decorators with TODOs to those cases that aren't supported on C++. Eventually everything will be supported everywhere and the skip decorators can go away.

Reply:
I am unsure I understand what you suggest.

I understand

```
Class Base:
   ...

class PythonJit(absltest.TestCase, Base)

  @property
  def jit(self):
    return _python_jit

class CPPJit(absltest.TestCase, Base)
  @unittest.SkipTest("Not supported for CPPjit")
  def test_someting(self):
```

where CPPJit overrides `test_someting` which is defined in Base.

I know I had some issue creating such base-class, and I would find it strange to duplicate all unsupported tests in CPPJit.

I am finding the current situation clear: The tests that both pass are in CPPJit.
The test that only the Python code pass, are in Pythonjit.

Reply:
I guess it's fine as is.